python/mxnet/gluon/rnn/rnn_cell.py

```diff
@@ -297,6 +297,63 @@ def __init__(self, prefix=None, params=None):
     def hybrid_forward(self, F, x, *args, **kwargs):
         raise NotImplementedError
 
+class _FusedBaseRNNCell(HybridRecurrentCell): # pylint: disable=abstract-method
+    """Implementation of recurrent layers."""
+    def __init__(self, hidden_size, input_size,
+                 i2h_weight_initializer, h2h_weight_initializer,
+                 i2h_bias_initializer, h2h_bias_initializer,
+                 mode, **kwargs):
+        super(_FusedBaseRNNCell, self).__init__(**kwargs)
+        self._hidden_size = hidden_size
+        self._input_size = input_size
+        self._mode = mode
+        num_gates = {'rnn_relu': 1, 'rnn_tanh': 1, 'lstm': 4, 'gru': 3}[mode]
+        self._gates = num_gates
+        self.i2h_weight = self.params.get('i2h_weight', shape=(num_gates*hidden_size, input_size),
+                                          init=i2h_weight_initializer,
+                                          allow_deferred_init=True)
+        self.h2h_weight = self.params.get('h2h_weight', shape=(num_gates*hidden_size, hidden_size),
+                                          init=h2h_weight_initializer,
+                                          allow_deferred_init=True)
+        self.i2h_bias = self.params.get('i2h_bias', shape=(num_gates*hidden_size,),
+                                        init=i2h_bias_initializer,
+                                        allow_deferred_init=True)
+        self.h2h_bias = self.params.get('h2h_bias', shape=(num_gates*hidden_size,),
+                                        init=h2h_bias_initializer,
+                                        allow_deferred_init=True)
+
+    def hybrid_forward(self, F, inputs, states, i2h_weight,
+                       h2h_weight, i2h_bias, h2h_bias):
+        prefix = 't%d_'%self._counter
+        params = F.concat(i2h_weight.reshape((-1,)),
+                          h2h_weight.reshape((-1,)),
+                          i2h_bias.reshape((-1,)),
+                          h2h_bias.reshape((-1,)), dim=0)
+        states = [s.reshape((-4, 1, -1, 0)) for s in states]
```

Comment:
We can change RNN operator to support single step mode. i.e. input and states are two dimensional instead of 3 dimensional